python/ray/tune/trainable.py

```diff
@@ -809,3 +811,20 @@ def _export_model(self, export_formats, export_dir):
 
     def _implements_method(self, key):
         return hasattr(self, key) and callable(getattr(self, key))
+
+
+class DistributedTrainable(Trainable):
```

Comment:
Actually do we need another layer of abstraction?

I'm thinking you can just implement this as a single helper function that you call to create the config.

After all, this is just providing a function (and not actually affecting the state of the trainable).

Reply:
Generally I think the layering of abstraction should be used only if we can group 2 or 3 functions together. 

The reason is two-fold:

1. Extra abstraction layering makes the code harder to dive into. As OSS, code should be easy to access and read and contribute to. it's good to keep a balance of organization vs readability and lean towards readability (not requiring new contributors to jump through too many layers). 
2. Aside from organization, there's no "functional" benefit for introducing this layer - it doesn't modify state, it doesn't introduce a new control flow, etc.

Reply:
cc @amogkam to consolidate discussion

Reply:
Realized these were all internal parameters.... not great but fine to merge